packages/flutter/lib/src/material/overscroll_indicator.dart

```diff
@@ -0,0 +1,214 @@
+          path.moveTo(0.0, size.height);
+          path.relativeLineTo(width, 0.0);
+          path.relativeLineTo(0.0, -rectBias);
+          path.relativeQuadraticBezierTo(width / -2.0, -arcBias, -width, 0.0);
+        }
+        break;
+      case Axis.horizontal:
+        final double height = size.height;
+        if (isLeading) {
+          path.moveTo(0.0, 0.0);
+          path.relativeLineTo(0.0, height);
+          path.relativeLineTo(rectBias, 0.0);
+          path.relativeQuadraticBezierTo(arcBias, height / -2.0, 0.0, -height);
+        } else {
+          path.moveTo(size.width, 0.0);
+          path.relativeLineTo(0.0, height);
+          path.relativeLineTo(-rectBias, 0.0);
+          path.relativeQuadraticBezierTo(-arcBias, height / -2.0, 0.0, -height);
+        }
+        break;
+    }
+    path.close();
+
+    final Paint paint = new Paint()..color = color;
+    canvas.drawPath(path, paint);
+  }
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    if (color.alpha == 0)
+      return;
+    paintIndicator(canvas, size);
+  }
+
+  @override
+  bool shouldRepaint(_Painter oldPainter) {
+    return oldPainter.scrollDirection != scrollDirection
+      || oldPainter.extent != extent
+      || oldPainter.isLeading != isLeading
+      || oldPainter.color != color;
+  }
+}
+
+/// When the child's Scrollable descendant overscrolls, displays a
+/// a translucent arc over the affected edge of the child.
+/// If the OverscrollIndicator's child has more than one Scrollable descendant
+/// the scrollableKey parameter can be used to identify the one to track.
+class OverscrollIndicator extends StatefulWidget {
+  OverscrollIndicator({ Key key, this.scrollableKey, this.child }) : super(key: key) {
+    assert(child != null);
+  }
+
+  final Key scrollableKey;
+  final Widget child;
+
+  @override
+  _OverscrollIndicatorState createState() => new _OverscrollIndicatorState();
+}
+
+class _OverscrollIndicatorState extends State<OverscrollIndicator> {
+  final AnimationController _extentAnimation = new AnimationController(
+    lowerBound: _kMinIndicatorExtent,
+    upperBound: _kMaxIndicatorExtent,
+    duration: _kIndicatorHideDuration
+  );
+
+  Timer _hideTimer;
+  Axis _scrollDirection;
+  double _scrollOffset;
+  double _minScrollOffset;
+  double _maxScrollOffset;
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  void _hide() {
+    _hideTimer?.cancel();
+    _hideTimer = null;
+    _extentAnimation.reverse();
+  }
+
+  void _updateState(ScrollableState scrollable) {
+    final ExtentScrollBehavior scrollBehavior = scrollable.scrollBehavior;
+    if (scrollBehavior is! ExtentScrollBehavior)
+      return;
+    _scrollDirection = scrollable.config.scrollDirection;
+    _scrollOffset = scrollable.scrollOffset;
+    _minScrollOffset = scrollBehavior.minScrollOffset;
+    _maxScrollOffset = scrollBehavior.maxScrollOffset;
+  }
+
+  void _onScrollStarted(ScrollableState scrollable) {
+    _updateState(scrollable);
```

Comment:
When the scrollOffset changes and we're overscrolling, the extent animation's value is updated:
```
  void _onScrollUpdated(ScrollableState scrollable) {
    final double value = scrollable.scrollOffset;
    if ((value < _minScrollOffset || value > _maxScrollOffset) &&
        ((value - _scrollOffset).abs() > kPixelScrollTolerance.distance)) {
      _hideTimer?.cancel();
      _hideTimer = new Timer(_kIndicatorTimeoutDuration, _hide);
      // Changing the animation's value causes an implicit setState().
      _extentAnimation.value = value < _minScrollOffset ? _minScrollOffset - value : value - _maxScrollOffset;
    }
    _updateState(scrollable);
  }
```
Likewise when the scroll ends and we "hide" the indicator.

Shouldn't that be enough?